modules/sops/templates/default.nix

```diff
@@ -0,0 +1,108 @@
+{ config, pkgs, lib, options, ... }:
+with lib;
+with lib.types;
+with builtins;
+let
+  cfg = config.sops;
+  secretsForUsers = lib.filterAttrs (_: v: v.neededForUsers) cfg.secrets;
+  users = config.users.users;
+  substitute = pkgs.writers.writePython3 "substitute" { }
+    (replaceStrings [ "@subst@" ] [ "${subst-pairs}" ] (readFile ./subs.py));
```

Comment:
This requires an extra derivation per substitute, would it not be better to just pass an argument to `substitute`?

Reply:
I agree that just pass an argument would be better, I really can't recall why I choose this weird method when writing this piece of code years ago.